poetry/installation/pip_installer.py

```diff
@@ -25,9 +25,14 @@ def install(self, package, update=False):
         if package.source_type == "legacy" and package.source_url:
             parsed = urlparse.urlparse(package.source_url)
             if parsed.scheme == "http":
+                if not parsed.username:
```

Comment:
With the changes requested below. The authentication part of the url will not appear here, so this check is not necessary.

However, this will be an issue since `pip` won't be aware of the credentials. So this is where we should modify the url to include the credentials, I think.